### ISSUE TITLE
Replica Set username bug

### ISSUE THREAD

**Issue by NizarBlond** [balderdashy/sails#2780]
I'm trying to use Replica Set on Compose.io mongo db and the wrong user name is being sent by Sails app, this is what I did:

	   mongo: {
			adapter: 'sails-mongo',
			user: 'webapp',
			password: 'somepass',
			database: 'app32343494',
			replSet: {
				servers: [{
					host: 'candidate.12.mongolayer.com',
					port: 10585
				}, {
					host: 'candidate.13.mongolayer.com',
					port: 10555
				}],
			}
		},

This is the reply from Compose support team:

		You need to check how your framework is building a user. I'm seeing the following in the logs:

		2015-03-06T14:55:41.986+0000 [conn217110] Failed to authenticate webapp@app32343494 with mechanism MONGODB-CR: AuthenticationFailed key mismatch
		2015-03-06T14:55:41.991+0000 [conn217108]  authenticate db: app32343494 { authenticate: 1, user: "webapp", nonce: "xxx", key: "xxx" }
		2015-03-06T14:55:41.991+0000 [conn217108] Failed to authenticate webapp@app32343494 with mechanism MONGODB-CR: AuthenticationFailed key mismatch

The user name is 'webapp@app32343494' instead of 'webapp'.

**Comment by matthewarkin** [comment]
@NizarBlond this sounds like a waterline issue https://github.com/balderdashy/waterline

In my Sails instance we set the database url to include the authentication so it ends up looking something like `url: 'mongodb://username:password@lighthouse.1.mongolayer.com:10235`

**Comment by sailsbot** [comment]
Thanks for posting, @NizarBlond.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
